src/SqlStreamStore.Postgres/PostgresStreamStore.ReadAll.cs

```diff
@@ -18,68 +23,363 @@ partial class PostgresStreamStore
+
+                    return await ReadTrustedMessages(fromPositionInclusive, messages[messages.Count - 1].Position, maxCount, prefetch, correlation, cancellationToken);
+                }
+            }
+
+            if(IsEnd(readAllResult.Count, maxCount))
+                return (readAllResult, true);
+
+            readAllResult.RemoveAt(maxCount);
+            return (readAllResult, false);
+        }
+
+        private async Task PollTransactions(
+            Guid correlation,
+            TxIdList transactionsInProgress,
+            CancellationToken cancellationToken)
+        {
+            Logger.TraceFormat(
+                "{correlation} Transactions {transactions} are in progress, so gaps might be filled, start comparing",
+                correlation,
+                transactionsInProgress);
+
+            bool stillInProgress;
+            int count = 1;
+            int delayTime = 10;
+            long totalTime = 0;
+
+            var sw = Stopwatch.StartNew();
+            do
+            {
+                stillInProgress = await ReadAnyTransactionsInProgress(transactionsInProgress, cancellationToken);
+                Logger.TraceFormat(
+                    "{correlation} Transactions still pending. Query 'ReadAnyTransactionsInProgress' took: {timeTaken}ms",
+                    correlation,
+                    sw.ElapsedMilliseconds);
+
+                if(_settings.GapHandlingSettings.AllowSkip
+                   && totalTime > _settings.GapHandlingSettings.PossibleAllowedSkipTime)
+                {
+                    Logger.ErrorFormat(
+                        "{correlation} Possible SKIPPED EVENT as we will stop waiting for in progress transactions! One of these transactions: {transactions} is in progress for longer then {totalTime}ms",
+                        correlation,
+                        transactionsInProgress,
+                        _settings.GapHandlingSettings.PossibleAllowedSkipTime);
+                    return;
+                }
+
+                if(totalTime > _settings.GapHandlingSettings.PossibleDeadlockTime)
+                {
+                    Logger.ErrorFormat(
+                        "{correlation} Possible DEADLOCK! One of these transactions: {transactions} is in progress for longer then {totalTime}ms",
+                        correlation,
+                        transactionsInProgress,
+                        _settings.GapHandlingSettings.PossibleDeadlockTime);
+                }
+
+
+                if(delayTime > 0)
+                {
+                    Logger.TraceFormat("{correlation} Delay 'PollTransactions' for {delayTime}ms",
+                        correlation,
+                        delayTime);
+                    await Task.Delay(delayTime, cancellationToken);
+                }
+
+                if(count % 5 == 0)
+                    delayTime += 10;
+
+                totalTime += sw.ElapsedMilliseconds;
```

Comment:
Difficile since this doesn't include the delay
This could work

- DelayCheck
- Db call
- Add time to totalTime
- Check on totalTime

I would delay before DB call because:
You have the delay anyway since you don't break out of it.
So even if there are no transactions pending => 10ms delay
So doing the delay first makes sure there is more chance the transactions are 'donzo'.

Only thing I see as a 'problem' is that the checks of the totalTime is never added to the totalTime.
We are speaking nanoseconds here. 

Opinion?